src/reactapp/vite.config.js

```diff
@@ -0,0 +1,27 @@
+import { defineConfig } from 'vite'
+import prefresh from '@prefresh/vite';
+
+export default defineConfig({
+    build: {
+        cssCodeSplit: false,
+        outDir:  "../view/frontend/web",
+        emptyOutDir: true,
+        lib: {
+            formats: ['es'],
+            entry: "src/index.jsx",
+            fileName: "react-checkout",
+            name: "react-checkout.js",
+        },
+    },
+    esbuild: {
+        jsxFactory: 'h',
+        jsxFragment: 'Fragment',
+        jsxInject: `import { h, Fragment } from 'preact'`,
+      },
+        plugins: [prefresh()],
+     resolve: {
+        alias: {
+            react: 'preact/compat',
```

Comment:
Currently preact is only aliased in production env, so when u would run `npm run build` it would set `env` in `/reactapp/config-overrides.js` to `'production'` or `'development'` for `npm run start`. This was done because it is easier to debug the full react package during development.

Could you add a conditional to maintain this behaviour into this config too?